src/libs/catalyst/AscentCatalyst.cxx

```diff
@@ -0,0 +1,178 @@
+#include <iostream>
+#include <fstream>
+#include <string>
+
+#include <ascent.h>
+
+#include <catalyst.h>
+#include <catalyst_conduit.hpp>
+#include <catalyst_conduit_blueprint.hpp>
+#include <catalyst_stub.h>
+
+#include <conduit.hpp>
+#include <conduit_cpp_to_c.hpp>
+
+#include "catalyst_impl_ascent.h"
+
+#ifdef _USE_MPI
+#include <mpi.h>
+#endif
+
+namespace detail{
+
+class Instance
+{
+protected:
+    Instance() = default;
+
+private:
+    //pointer to the ascent instance
+    void* ascent = nullptr;
+    std::vector<conduit::Node> actions;
+    //instance pointer;
+    static Instance* instance;
+
+public:
+    // deleting copy constructor
+    Instance(const Instance& obj) = delete;
+
+    static Instance* GetInstance()
+    {
+      if(instance == nullptr)
+      {
+        instance = new Instance();
+        return instance;
+      }
+      else
+      {
+        return instance;
+      }
+    }
+
+    void SetAscent(void* _ascent) { this->ascent = _ascent; }
+    void* GetAscent() { return this->ascent; }
+
+    void AddActions(conduit::Node& node)
+    {
+      actions.push_back(node);
+    }
+    std::vector<conduit::Node> GetActions()
+    {
+      return this->actions;
+    }
+
+    void ClearActions()
+    {
+      this->actions.clear();
+    }
+};
+
+}
+
+detail::Instance* detail::Instance::instance = nullptr;
+
+enum catalyst_status catalyst_initialize_ascent(const conduit_node* params)
+{
+  std::cout << "[pre] Executing Initialize" << std::endl;
+  //Convert params to C++ node:
+  const conduit_cpp::Node cpp_params = conduit_cpp::cpp_node(const_cast<conduit_node*>(params));
+  cpp_params.print();
+  auto instance = detail::Instance::GetInstance();
+  void* ascent = ascent_create();
+  if(!ascent)
+  {
+    std::cout << "Error creating ascent" << std::endl;
+  }
+  instance->SetAscent(ascent);
+  // Assuming params contain information to initialize ascent
+  conduit::Node options;
+#ifdef _USE_MPI
+  options["mpi_comm"] = MPI_Comm_c2f(MPI_COMM_WORLD);
+#endif
+  options["runtime/type"] = "ascent";
+  conduit_node* options_c = conduit::c_node(&options);
+  ascent_open(ascent, options_c);
+
+  if (cpp_params.has_path("catalyst/scripts"))
+  {
+    auto& scripts = cpp_params["catalyst/scripts"];
+    conduit_index_t nchildren = scripts.number_of_children();
+    for (conduit_index_t i = 0; i < nchildren; ++i)
+    {
+      auto script = scripts.child(i);
+      const auto fname =
+        script.dtype().is_string() ? script.as_string() : script["filename"].as_string();
+      std::cout << " Provided script : " << fname << std::endl;
+      std::ifstream ifs(fname);
+      std::string content;
+      content.assign((std::istreambuf_iterator<char>(ifs)),
+                     (std::istreambuf_iterator<char>()));
+      conduit::Node node = conduit::Node();
+      node.parse(content, "yaml");
+      node.print();
```

Comment:
Is it for debug purpose? Should it be removed in the production version?

Reply:
Yes, it was intended for debug, I've removed it